### ISSUE TITLE
Add/update THP doc

### ISSUE THREAD

**Issue by Feediver1** [redpanda-data/documentation#1318]
Do we do anything around transparent huge pages? Do we use a page cache outside of the non-use for producer acks?

[Slack Message](https://redpandadata.slack.com/archives/C0246LPDYDU/p1677594379078119?thread_ts=1677594379.078119&cid=C0246LPDYDU)

**Comment by Feediver1** [comment]
Brandon Allard
Some changes folks have recommended recently that could improve performance;
Look into using huge pages in cloudv1/v2. See https://redpandadata.slack.com/archives/C044RD18NMV/p1669719220634009
Use ENA Express in AWS cloudv1/v2 to reduce network latency. See https://redpandadata.slack.com/archives/C01E8GRCMM4/p1669697060735899

Thanks, so we use THP in some cases automatically under the hood but it's not something that's tunable yet, is that the right takeaway?

...
is it recommended to disable THP in the OS?

62 posts in this string...

**Comment by ballard26** [comment]
Hey! Great questions. We do nothing with ENA express yet as support for it is pretty limited currently and none of our usual instance types and/or sizes are supported; see [here](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ena-express.html#ena-express-supported-instance-types). It is something I'll try to keep an eye on though.

As for THP. Since the seastar memory allocator [here](https://github.com/scylladb/seastar/blob/master/src/core/memory.cc#L1103) calls `::madvise(base, size, MADV_HUGEPAGE)` on the pages we allocate from for all memory in RP we should be using huge pages for everything. That being said seastar is only requesting huge pages. It's still possible to disable them in the kernel with something like 
```
echo never >/sys/kernel/mm/transparent_hugepage/enabled
````
I have confirmed that we had the kernel configured properly in cloudv1 and that huge pages were being used for all allocations. However, I haven't confirmed it in cloudv2. 

So in short, for THP, we always try to use them in the application if possible. We do not currently plan on making it tunable through RP. And I'm not sure of what one would gain by disabling it. But if a user wanted to tune things they can do so through the kernel.
